public/components/common/search/sql_search.tsx

```diff
@@ -0,0 +1,305 @@
+export const SQLSearch = (props: any) => {
+  const {
+    query,
+    tempQuery,
+    handleQueryChange,
+    dslService,
+    selectedPanelName,
+    selectedCustomPanelOptions,
+    setSelectedPanelName,
+    setSelectedCustomPanelOptions,
+    handleSavingObject,
+    isPanelTextFieldInvalid,
+    savedObjects,
+    showSavePanelOptionsList,
+    showSaveButton = true,
+    selectedSubTabId,
+    searchBarConfigs = {},
+    getSuggestions,
+    onItemSelect,
+    tabId = '',
+    baseQuery = '',
+    curVisId,
+    setSubType,
+    setIsQueryRunning,
+  } = props;
+
+  const explorerSearchMetadata = useSelector(selectSearchMetaData)[tabId];
+  const dispatch = useDispatch();
+  const appLogEvents = tabId.match(APP_ANALYTICS_TAB_ID_REGEX);
+  const [isSavePanelOpen, setIsSavePanelOpen] = useState(false);
+  const [isFlyoutVisible, setIsFlyoutVisible] = useState(false);
+  const [queryLang, setQueryLang] = useState([{ label: explorerSearchMetadata.lang }]);
+  const [jobId, setJobId] = useState('');
+  const sqlService = new SQLService(coreRefs.http);
+
+  const {
+    data: pollingResult,
+    loading: pollingLoading,
+    error: pollingError,
+    startPolling,
+    stopPolling,
+  } = usePolling<any, any>((params) => {
+    return sqlService.fetchWithJobId(params);
+  }, 5000);
+
+  const requestParams = { tabId };
+  const { getLiveTail, getEvents, getAvailableFields, dispatchOnGettingHis } = useFetchEvents({
+    pplService: new SQLService(coreRefs.http),
+    requestParams,
+  });
+
+  const closeFlyout = () => {
+    setIsFlyoutVisible(false);
+  };
+
+  const showFlyout = () => {
+    setIsFlyoutVisible(true);
+  };
+
+  let flyout;
+  if (isFlyoutVisible) {
+    flyout = <PPLReferenceFlyout module="explorer" closeFlyout={closeFlyout} />;
+  }
+
+  const Savebutton = (
+    <EuiButton
+      iconSide="right"
+      onClick={() => {
+        setIsSavePanelOpen((staleState) => {
+          return !staleState;
+        });
+      }}
+      data-test-subj="eventExplorer__saveManagementPopover"
+      iconType="arrowDown"
+    >
+      Save
+    </EuiButton>
+  );
+
+  const handleQueryLanguageChange = (lang) => {
+    if (lang[0].label === 'DQL') {
+      window.location.href = '/app/data-explorer/discover';
+      return;
+    }
+    dispatch(
+      updateSearchMetaData({
+        tabId,
+        data: { lang: lang[0].label },
+      })
+    );
+    setQueryLang(lang);
+  };
+
+  const onQuerySearch = (lang) => {
+    setIsQueryRunning(true);
+
+    sqlService
+      .fetch({
+        lang: lowerCase(lang[0].label),
+        query: tempQuery,
+      })
+      .then((result) => {
+        if (result.queryId) {
+          setJobId(result.queryId);
+          startPolling({
+            queryId: result.queryId,
+          });
+        } else {
+          console.log('no query id found in response');
+        }
+      })
+      .catch((e) => {
+        setIsQueryRunning(false);
+        console.error(e);
+      })
+      .finally(() => {});
```

Comment:
same question here, remove empty finally and how does user know when error happened?